File: 13. Roman to Integer/solution.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        dic = {
            "I": 1,
            "V": 5,
            "X": 10,
            "L": 50,
            "C": 100,
            "D": 500,
            "M": 1000
            }

        sum = 0
        i = 0
        for i in range(len(s)):
            if i + 1 < len(s) and dic[s[i]] < dic[s[i + 1]]:
                sum -= dic[s[i]]
            else:
                sum += dic[s[i]]
        return sum

File: 13. Roman to Integer/solution.js
/**
 * @param {string} s
 * @return {number}
 */
var romanToInt = function (s) {
    const roman_dic = {
        "I": 1,
        "V": 5,
        "X": 10,
        "L": 50,
        "C": 100,
        "D": 500,
        "M": 1000
    }
    let sum = 0;
    for (let i = 0; i < s.length; i++) {
        if (i + 1 < s.length && roman_dic[s[i]] < roman_dic[s[i + 1]]) {
            sum -= roman_dic[s[i]]
        } else {
            sum += roman_dic[s[i]]
        }
    }
    return sum
};